File: Downloads/project-bolt-sb1-uc7pp9or/project/scripts/setupCodes.ts
import { connectToDatabase } from '../src/lib/mongodb';
import { Db } from 'mongodb';

interface ShopCode {
  code: string;
  type: 'regular' | 'top_score';
  description: string;
  expiryDate: Date;
  isActive: boolean;
  createdAt: Date;
}

const codeSchema = {
  validator: {
    $jsonSchema: {
      bsonType: 'object',
      required: ['code', 'type', 'description', 'expiryDate', 'isActive'],
      properties: {
        code: { bsonType: 'string' },
        type: { enum: ['regular', 'top_score'] },
        description: { bsonType: 'string' },
        expiryDate: { bsonType: 'date' },
        isActive: { bsonType: 'bool' },
        createdAt: { bsonType: 'date' }
      }
    }
  }
};

async function setupCodes() {
  try {
    const db: Db = await connectToDatabase();
    console.log('Connected to MongoDB');

    // Create codes collection if it doesn't exist
    if (!db.collection('codes')) {
      await db.createCollection('codes', codeSchema);
    }

    const codesCollection = db.collection<ShopCode>('codes');

    // First, deactivate all existing codes
    await codesCollection.updateMany({}, { $set: { isActive: false } });

    // Create regular code
    await codesCollection.insertOne({
      code: 'HHNYC2024',
      type: 'regular',
      description: "Thanks for playing! Here's your code to use in our shop.",
      expiryDate: new Date('2024-12-31'),
      isActive: true,
      createdAt: new Date()
    });

    // Create top score code
    await codesCollection.insertOne({
      code: 'HHNYC2024PRO',
      type: 'top_score',
      description: "Congratulations on your amazing score! Here's your special shop code.",
      expiryDate: new Date('2024-12-31'),
      isActive: true,
      createdAt: new Date()
    });

    console.log('Codes setup completed successfully!');
    process.exit(0);
  } catch (error) {
    console.error('Error setting up codes:', error);
    process.exit(1);
  }
}

setupCodes();

File: Downloads/project-bolt-sb1-uc7pp9or/project/scripts/SetupCodes.java
import com.mongodb.client.*;
import com.mongodb.MongoClientSettings;
import com.mongodb.ConnectionString;
import org.bson.Document;
import java.util.Date;
import java.util.Calendar;

public class SetupCodes {
    public static void main(String[] args) {
        try {
            // Connect to MongoDB
            MongoClient mongoClient = MongoClients.create(System.getenv("MONGODB_URI"));
            MongoDatabase database = mongoClient.getDatabase("nyc-subway-trivia");
            MongoCollection<Document> codesCollection = database.getCollection("codes");

            // First, deactivate all existing codes
            codesCollection.updateMany(
                new Document(),
                new Document("$set", new Document("isActive", false))
            );

            // Set expiry date to December 31, 2024
            Calendar cal = Calendar.getInstance();
            cal.set(2024, Calendar.DECEMBER, 31);
            Date expiryDate = cal.getTime();

            // Create regular code
            Document regularCode = new Document()
                .append("code", "HHNYC2024")
                .append("type", "regular")
                .append("description", "Thanks for playing! Here's your code to use in our shop.")
                .append("expiryDate", expiryDate)
                .append("isActive", true)
                .append("createdAt", new Date());

            codesCollection.insertOne(regularCode);

            // Create top score code
            Document topScoreCode = new Document()
                .append("code", "HHNYC2024PRO")
                .append("type", "top_score")
                .append("description", "Congratulations on your amazing score! Here's your special shop code.")
                .append("expiryDate", expiryDate)
                .append("isActive", true)
                .append("createdAt", new Date());

            codesCollection.insertOne(topScoreCode);

            System.out.println("Codes setup completed successfully!");
            mongoClient.close();
        } catch (Exception e) {
            System.err.println("Error setting up codes: " + e.getMessage());
            System.exit(1);
        }
    }
}
